components/payments/settings/payments-webhooks-table.tsx

```diff
@@ -0,0 +1,311 @@
+      </FormControl>
+    );
+  },
+);
+
+interface EditTableRowProps {
+  webhook: PaymentsWebhooksType;
+  onEdit: (webhook: PaymentsWebhooksType, newUrl: string) => void;
+  onDelete: (webhook: PaymentsWebhooksType) => void;
+}
+
+const EditTableRow = ({ webhook, onEdit, onDelete }: EditTableRowProps) => {
+  const [editUrl, setEditUrl] = useState(webhook.url ?? "");
+  const [isEdit, setIsEdit] = useState(false);
+
+  const editWebhookRef = useRef<HTMLInputElement>(null);
+
+  const startEdit = () => {
+    setIsEdit(true);
+    setEditUrl(webhook.url);
+    if (editWebhookRef.current) {
+      editWebhookRef.current.focus();
+      editWebhookRef.current.select();
+    }
+  };
+
+  const acceptEdit = () => {
+    onEdit(webhook, editUrl);
+    setIsEdit(false);
+  };
+
+  const rejectEdit = () => {
+    setIsEdit(false);
+  };
+
+  const _onDelete = () => {
+    onDelete(webhook);
+  };
+
+  return (
+    <Tr borderBottomWidth={1} _last={{ borderBottomWidth: 0 }}>
+      <Td borderBottomWidth="inherit" borderBottomColor="accent.100">
+        {isEdit ? (
+          <UrlInputField
+            webhookUrl={editUrl}
+            onChange={setEditUrl}
+            ref={editWebhookRef}
+          />
+        ) : (
+          <Text>{webhook.url}</Text>
+        )}
+      </Td>
+      <Td borderBottomWidth="inherit" borderBottomColor="accent.100">
+        {isEdit ? (
+          <ButtonGroup variant="ghost" gap={2}>
+            <IconButton
+              onClick={acceptEdit}
+              variant="outline"
+              icon={<Icon as={FaCheck} boxSize={4} />}
+              aria-label="Accept"
+            />
+            <IconButton
+              onClick={rejectEdit}
+              variant="outline"
+              icon={<Icon as={FaXmark} boxSize={4} />}
+              aria-label="Cancel"
+            />
+          </ButtonGroup>
+        ) : (
+          <ButtonGroup variant="ghost" gap={2}>
+            <IconButton
+              onClick={startEdit}
+              variant="outline"
+              icon={<Icon as={BiPencil} boxSize={4} />}
+              aria-label="Edit"
+            />
+            <IconButton
+              onClick={_onDelete}
+              variant="outline"
+              icon={<Icon as={FiTrash2} boxSize={4} />}
+              aria-label="Remove"
+            />
+          </ButtonGroup>
+        )}
+      </Td>
+    </Tr>
+  );
+};
+
+interface CreateTableRowProps {
+  onCreate: (webhookUrl: string) => void;
+}
+
+const CreateTableRow = ({ onCreate }: CreateTableRowProps) => {
+  const [createUrl, setCreateUrl] = useState("");
+  const createWebhookRef = useRef<HTMLInputElement>(null);
+
+  const _onCreate = () => {
+    onCreate(createUrl);
+    setCreateUrl("");
+  };
+
+  return (
+    <Tr borderBottomWidth={1} _last={{ borderBottomWidth: 0 }}>
+      <Td borderBottomWidth="inherit" borderBottomColor="accent.100">
+        <UrlInputField
+          webhookUrl={createUrl}
+          onChange={setCreateUrl}
```

Comment:
see here would makes more sense to use useForm even if it's a simple thing (and then we don't need useState) and we let useForm handle all the logic and validation